cpp/core/shuffle/utils.h

```diff
@@ -106,8 +107,16 @@ static inline arrow::Result<std::string> createTempShuffleFile(const std::string
     ARROW_ASSIGN_OR_RAISE(auto file_info, fs->GetFileInfo(filePath));
     if (file_info.type() == arrow::fs::FileType::NotFound) {
       exist = false;
-      ARROW_ASSIGN_OR_RAISE(auto s, fs->OpenOutputStream(filePath));
-      RETURN_NOT_OK(s->Close());
+      int fd = open(filePath.c_str(), O_CREAT | O_EXCL | O_RDWR, 0666);
+      if (fd < 0) {
+        if (errno == EEXIST) {
+          exist = true;
```

Comment:
Hmm, in which case it will return ```EEXIST``` given we have ```file_info.type() == arrow::fs::FileType::NotFound``` was true.

https://arrow.apache.org/docs/cpp/api/filesystem.html#_CPPv4N5arrow2fs8FileTypeE

Reply:
`file_info.type() == arrow::fs::FileType::NotFound` only make sure the file doesn't exist, but it can't make sure the file is being created at the same time by other threads.